lib/grape/validations/params_scope.rb

```diff
@@ -232,16 +232,25 @@ def validates(attrs, validations)
         default = validations[:default]
         doc_attrs[:default] = default if validations.key?(:default)
 
-        values = options_key?(:values, :value, validations) ? validations[:values][:value] : validations[:values]
+        values = validations[:values]
+        if values.is_a? Hash
+          excepts = values[:except]
+          values = values[:value]
+          values = nil if values.is_a? Proc
```

Comment:
This is a little hard to read. Maybe flatten this thing, if I understand correctly you can do something like this:

```
values = options_key?(:values, :value, validations) ? validations[:values][:value] : validations[:values]
excepts = ...
```

Give it a try?